frontend/app/ui_components/date/date-range-picker-directive.js

```diff
@@ -114,24 +129,25 @@ module.exports = function(TimezoneService, ConfigurationService,
           $timeout.cancel(endTimerId);
           return;
         }
-        $timeout.cancel(startTimerId);
-        startTimerId = $timeout(function() {
+        $timeout.cancel(endTimerId);
+        endTimerId = $timeout(function() {
           var date = inputEnd.val(),
-              isValid = TimezoneService.isValid(date, 'DD/MM/YYYY');
+              isValid = TimezoneService.isValid(date, customDateFormat);
 
           if(isValid){
-            scope.endDate = formattedISODate(date);
+            scope.endDate = formattedISODate(parseDate(date, customDateFormat));
           }
         }, 1000);
+      }).on('click', function(){
+        inputEnd.datepicker('show');
       });
 
       inputStart.datepicker({
         firstDay: ConfigurationService.startOfWeek(),
         showWeeks: true,
         changeMonth: true,
-        numberOfMonths: 2,
         showButtonPanel: true,
-        dateFormat: 'dd/mm/yy',
+        dateFormat: datepickerFormat,
```

Comment:
@NobodysNightmare, example is there. And documentation also https://api.jqueryui.com/datepicker/#option-dateFormat. And also http://api.jqueryui.com/datepicker/#utility-formatDate, jQueryUI uses formatDate in their behind.